subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsBuildOperationResult.java

```diff
@@ -71,6 +75,39 @@ public SnapshotTaskInputsBuildOperationResult(CachingState cachingState) {
         this.cachingState = cachingState;
     }
 
+    private static final String FINGERPRINTING_STRATEGY_PREFIX = "FINGERPRINTING_STRATEGY_";
+    private static final String DIRECTORY_SENSITIVITY_PREFIX = "DIRECTORY_SENSITIVITY_";
+    private static final String LINE_ENDING_SENSITIVITY_PREFIX = "LINE_ENDING_SENSITIVITY_";
+
+    enum PropertyAttribute {
```

Comment:
Worth some documentation to indicate that the names here, once used, can't change.